dialogue-core/src/main/java/com/palantir/dialogue/core/LimitedChannel.java

```diff
@@ -26,6 +26,9 @@
  * Similar to {@link com.palantir.dialogue.Channel}, but may not actually execute the call (eg: when the channel is
  * overloaded). Semantics match {@link com.palantir.dialogue.Channel} aside from returning an
  * {@link Optional optional response}.
+ * Limited channels must limit exclusively based on the state of the {@link com.palantir.dialogue.Channel}, not
+ * the {@link Endpoint} or {@link Request} arguments, otherwise the caller (generally a {@link QueuedChannel})
```

Comment:
Given our conversation this morning, shall we also update the blacklisting channel to adhere to this spec change?

Reply:
Might as well :-)

Reply:
I'm a little sad about losing the per endpoint blacklisting tbh :/

Reply:
We can get it back, but we'll need a separate queueing + limiting mechanism. For what it's worth, I think we want the ability to both limit entire channels as well as per-endpoint separately using similar mechanisms triggered with different configurations (rate of failure, response codes, duration of the blacklist etc)

Reply:
I figured we'd pull both for now, we can do something else if you like.